modules/install/pages/upgrade-cluster-online-full-capacity.adoc

```diff
@@ -0,0 +1,236 @@
+The `REPO_NAME` is the name of the repository that will be associated with the location.
+
+Once the archive and repository have been created, a command such as the following performs a full backup:
+
+[source,bash]
+----
+cbbackupmgr backup --archive ${ABS_PATH_TO_ARCHIVE} --repo ${REPO_NAME} \
+--cluster ${CLUSTER_ADDRESS} --username ${USERNAME} --password ${PASSWORD} \
+--full-backup
+----
+
+Here, the `CLUSTER_ADDRESS` is the IP address or domain name of the cluster that is being backed up.
+The `full-backup` flag ensures that the backup is indeed a _full_ backup.
+
+For the equivalent procedure as performed by the Backup Service, see xref:manage:manage-backup-and-restore/manage-backup-and-restore.adoc#run-an-immediate-backup[Run an Immediate Backup].
+
+In the event of node-upgrade failing, the backed-up data can be restored by means of xref:backup-restore:cbbackupmgr-restore.adoc[cbbackupmgr restore] or xref:cli:cbtools/cbrecovery.adoc[cbrecovery]; or by means of the Backup Service (see xref:manage:manage-backup-and-restore/manage-backup-and-restore.adoc#restore-backups[Restore Backups]).
+
+[#add-and-remove-a-node]
+== Stage Three: Add the Spare Node, and Remove a Node
+
+_Add_ the upgraded, removed node back into the cluster, and _remove_ a node that is currently part of the cluster.
+The node that is being added should be configured to run the same service as that running on the node that is to be removed.
+For example, if the node to be removed is running the Data Service, configure the node to be added to run the Data Service.
+Couchbase Server will execute the rebalance as a _swap rebalance_, to maximize efficiency.
+
+For an overview of node-removal, see xref:learn:clusters-and-availability/removal.adoc[Removal]; and for practical examples of performing removal, see xref:manage:manage-nodes/remove-node-and-rebalance.adoc.
+For an overview of node-addition, see xref:learn:clusters-and-availability/nodes.html#clusters[Clusters]; and for practical examples of node-addition, see xref:manage:manage-nodes/add-node-and-rebalance.adoc[Add a Node and Rebalance].
+
+Note that for the CLI and REST API, the staging of a swap rebalance requires _two_ separate commands.
+The first specifies that one or more nodes be _added_ to the cluster: however, this command requires a subsequent rebalance to be performed, to complete the process.
+The second command indeed specifies that subsequent rebalance, but also includes an instruction to _remove_ one or more nodes from the cluster: therefore, as the rebalance occurs, it finalizes both node-addition and node-removal.
+
+[#upgrade-the-removed-node]
+== Stage Four: Upgrade the Removed Node
+
+Upgrade the node that has just been removed, by proceeding as follows:
+
+. Stop the `couchbase-server.service` service, on the node.
+Enter the following command:
++
+----
+systemctl stop couchbase-server.service
```

Comment:
I think this section has been repeated, could it be extracted into a template document?

(continues all the way past the install procedure. If there's specifics, we could avoid duplication by using conditionals in the template)

Reply:
Yes, once we have the procedures settled, I will pull stuff out into asciidoc partials. We may want to handle non-root before that, and I think we'll also need to look at commands for some additional platforms, including Windows.